src/Microsoft.Extensions.ProjectModel.Sources/DotNet/DotNetProjectContextFactory.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.DotNet.ProjectModel;
+
+namespace Microsoft.Extensions.ProjectModel
+{
+    internal class DotNetProjectContextFactory
+    {
+        public IProjectContext Create(string filePath, string configuration)
+        {
+            var project = new ProjectContextBuilder()
+                .AsDesignTime()
```

Comment:
IIUC this API ensures project.json => MSBuild references show up in list of dependencies.